api-2.0/src/test/java/org/openmrs/module/attachments/AttachmentsServiceTest.java

```diff
@@ -24,36 +27,56 @@
 	private AttachmentsService as;
 	
 	@Autowired
-	protected TestHelper testHelper;
+	private TestHelper testHelper;
 	
 	@Autowired
 	@Qualifier("patientService")
-	protected PatientService ps;
+	private PatientService ps;
 	
 	@Autowired
 	@Qualifier("visitService")
-	protected VisitService vs;
+	private VisitService vs;
 	
 	@Autowired
 	@Qualifier("encounterService")
 	private EncounterService es;
 	
+	@Autowired
+	@Qualifier("conceptService")
+	private ConceptService cs;
+	
+	@Autowired
+	@Qualifier("obsService")
+	private ObsService os;
+	
 	@Test
 	public void getAttachments_shouldReturnAttachments() throws Exception {
 		
 		// Setup
-		List<Obs> obsList = testHelper.saveComplexObs(3);
+		List<Obs> complexObsList = testHelper.saveComplexObs(3);
 		
-		Obs obs = obsList.get(0);
+		Obs obs = complexObsList.get(0);
 		Patient patient = obs.getEncounter().getPatient();
 		Visit visit = obs.getEncounter().getVisit();
 		Encounter encounter = obs.getEncounter();
 		
+		// saving some other obs during the same encounter
+		Obs otherObs = new Obs();
+		otherObs.setConcept(cs.getConcept(3));
+		otherObs.setObsDatetime(new Date());
+		otherObs.setEncounter(encounter);
+		otherObs.setPerson(patient);
+		otherObs.setValueText("Some text value for a test obs.");
+		otherObs = os.saveObs(otherObs, null);
+		Assert.assertNotNull(otherObs.getId());
+		
+		complexObsList.add(otherObs);
```

Comment:
@dilekamadushan the point was to show that the sole existence of a non-complex obs would fail your test. You **cannot** add the non-complex obs to the list of complex obs.

Without that line the test is "correct", albeit failing, because of the service implementation is inadequate.

Reply:
@mks-d thanks. I added a separate test method and modified the AttachemntServiceImpl to check for non-complex Obs.